### PR TITLE
feat(localstorev2): subscribe push

### PR DESCRIPTION
**Pull request by notanatol**
### Checklist

- [X] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).

### Description
Rebased push subscriber functionality

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3853)

### CODE REVIEW

File: pkg/localstorev2/storer.go [hunk 1]
```diff
@@ -370,6 +374,138 @@ func (p *putterSession) Done(addr swarm.Address) error { return p.done(addr) }
 
 func (p *putterSession) Cleanup() error { return p.cleanup() }
 
+func (db *DB) SubscribePush(ctx context.Context) (chunks chan swarm.Chunk, reset, stop func()) {
```
Comment (istae) [inline pkg/localstorev2/storer.go:377]: this should ideally be in it's own file

File: pkg/localstorev2/storer.go [hunk 2]
```diff
@@ -370,6 +374,138 @@ func (p *putterSession) Done(addr swarm.Address) error { return p.done(addr) }
...
+	db.pushTriggers = append(db.pushTriggers, trigger)
+	db.pushTriggersMu.Unlock()
+
+	stopChan := make(chan struct{})
+	var stopChanOnce sync.Once
+
+	var sinceItem swarm.Chunk
+
+	reset = func() {
+		time.Sleep(1 * time.Second) // give some time when retrying
```
Comment (istae) [inline pkg/localstorev2/storer.go:395]: why do we sleep?

File: pkg/localstorev2/storer.go [hunk 3]
```diff
@@ -370,6 +374,138 @@ func (p *putterSession) Done(addr swarm.Address) error { return p.done(addr) }
...
 
+func (db *DB) SubscribePush(ctx context.Context) (chunks chan swarm.Chunk, reset, stop func()) {
+	chunks = make(chan swarm.Chunk)
+	trigger := make(chan struct{}, 1)
+	resetC := make(chan struct{}, 1)
+
+	// send signal for the initial iteration
+	trigger <- struct{}{}
+
+	db.pushTriggersMu.Lock()
```
Comment (istae) [inline pkg/localstorev2/storer.go:385]: let's make use of the ./pkg/localstore/internal/events pkg (it's in localstorev2-reserve branch atm)

File: pkg/localstorev2/storer.go (hunk 2 above)
Comment (notanatol) [reply pkg/localstorev2/storer.go:395]: no sure, this functionality was ported from the old subscriber

File: pkg/localstorev2/storer.go (hunk 3 above)
Comment (notanatol) [reply pkg/localstorev2/storer.go]: let's pair on this one, couldn't get far with it

File: pkg/localstorev2/storer.go (hunk 2 above)
Comment (istae) [reply pkg/localstorev2/storer.go]: with the rewrite, we have a good opportunity to step back, reconsider things with hindsight. I would advocate for a redesign  between the upload - pusher interactions IF the old mechanism are not up to par.

**Pull request closed without merging**